app/src/main/java/com/danielpriddle/drpnews/App.kt

```diff
@@ -0,0 +1,43 @@
+package com.danielpriddle.drpnews
+
+import android.app.Application
+import android.net.ConnectivityManager
+import androidx.datastore.preferences.preferencesDataStore
+import com.danielpriddle.drpnews.data.database.NewsDatabase
+import com.danielpriddle.drpnews.data.networking.buildApiService
+import com.danielpriddle.drpnews.data.repository.ArticleRepository
+import com.danielpriddle.drpnews.data.repository.ArticleRepositoryImpl
+import com.danielpriddle.drpnews.data.networking.APINewsService
+import com.danielpriddle.drpnews.data.preferences.PreferencesDataStoreImpl
+import com.danielpriddle.drpnews.utils.NetworkStatusChecker
+
+class App : Application() {
+    val dataStore by preferencesDataStore(name = "preferences")
+
+    companion object {
+        private lateinit var instance: App
+        private val database: NewsDatabase by lazy {
+            NewsDatabase.buildDatabase(instance)
+        }
+        private val newsService by lazy {
+            APINewsService(buildApiService(),
+                NetworkStatusChecker(instance.getSystemService(ConnectivityManager::class.java)))
```

Comment:
really minor, but typical formatting for this would put each param on new lines:

```suggestion
            APINewsService(
                buildApiService(),
                NetworkStatusChecker(instance.getSystemService(ConnectivityManager::class.java)),
            )
```